### PR TITLE
#580 evasion tests integration_C

### LINKED ISSUE #580: Move all Evasion tests to the new ART Testing Framework and make them framework independent
**Is your feature request related to a problem? Please describe.**
Most evasion tests are still running as framework specific tests, which contributes to reaching the Travis test limit.

**Describe the solution you'd like**
Move all Evasion tests to the new ART Testing Framework and make them framework independent, this will significantly reduce the time required to run all the tests on Travis

### PR DESCRIPTION
**Pull request by killianlevacher**
# Description

This PR addresses issue #580 which is about integrating remaining evasion tests within the new pytest framework

Fixes #580 evasion

## Type of change

Please check all relevant options.

- [X] Improvement (non-breaking)
- [ ] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: conftest.py [hunk 1]
```diff
@@ -138,7 +138,7 @@ def _image_dl_estimator_for_attack(attack, defended=False, **kwargs):
             raise ARTTestFixtureNotImplemented(
                 "no estimator available", image_dl_estimator_for_attack.__name__, framework, {"attack": attack}
             )
-        return classifier_tested[0]
+        return classifier_tested[0], sess
```
Comment (beat-buesser) [inline conftest.py:141]: Why is this change required? The returned `sess` does not seem to be used anywhere.

**Review by beat-buesser — CHANGES_REQUESTED** [review]

File: conftest.py (hunk 1 above)
Comment (killianlevacher) [reply conftest.py:141]: ye it's rarely used indeed but the session is needed some times for Tensorflow models. That's a discrepancy that I hadn't seen before. The session is used in some test cases when `image_dl_estimator` is called directly, hence `image_dl_estimator_for_attack` which is just an augmented version of  `image_dl_estimator` by convention should have passed on that session too from the start. That was something I had overlooked so I just felt I should fix this now rather than later when someone complains why isn't it passed on :-)
